src/ring_theory/coprime.lean

```diff
@@ -0,0 +1,74 @@
+/-
+Copyright (c) 2020 Kenny Lau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kenny Lau, Ken Lee, Chris Hughes
+-/
+
+import algebra.big_operators
+import data.fintype.basic
+import data.set.disjointed
+
+/-!
+# Coprime elements of a ring
+
+## Main definitions
+
+* `is_coprime x y`: that `x` and `y` are coprime, defined to be the existence of `a` and `b` such
+that `a * x + b * y = 1`.
+
+-/
+
+open_locale classical big_operators
+
+universes u v
+
+variables {R : Type u} [comm_semiring R] (x y z : R)
+
+/-- The proposition that `x` and `y` are coprime, defined to be the existence of `a` and `b` such
+that `a * x + b * y = 1`. -/
+@[simp] def is_coprime : Prop :=
+∃ a b, a * x + b * y = 1
+
+variables {x y z}
+
+theorem is_coprime_self :
+  is_coprime x x ↔ is_unit x :=
+⟨λ ⟨a, b, h⟩, is_unit_of_mul_eq_one x (a + b) $ by rwa [mul_comm, add_mul],
+λ h, let ⟨b, hb⟩ := is_unit_iff_exists_inv'.1 h in ⟨b, 0, by rwa [zero_mul, add_zero]⟩⟩
+
+theorem is_coprime.mul_dvd (H : is_coprime x y) (H1 : x ∣ z) (H2 : y ∣ z) : x * y ∣ z :=
+begin
+  obtain ⟨a, b, h⟩ := H,
+  rw [← mul_one z, ← h, mul_add],
+  apply dvd_add,
+  { rw [mul_comm z, mul_assoc],
+    exact dvd_mul_of_dvd_right (mul_dvd_mul_left _ H2) _ },
+  { rw [mul_comm b, ← mul_assoc],
+    exact dvd_mul_of_dvd_left (mul_dvd_mul_right H1 _) _ }
+end
+
+variables {I : Type v} {s : I → R}
+
+theorem is_coprime_prod_of_pairwise_coprime (hs : pairwise (is_coprime on s)) {t : finset I}
+  {x : I} : x ∉ t → is_coprime (s x) (∏ y in t, s y) :=
```

Comment:
A better lemma would be
```lean
theorem is_coprime.prod_right {t : finset I} {a : R} (h : ∀ i ∈ t, is_coprime a (s i)) : is_coprime a (∏ i in t, s i) := sorry
```

It should be preceded by
```
theorem is_coprime.one_right (a : R) : is_coprime a 1 := sorry
theorem is_coprime.mul_right {a b c : R} (hab : is_coprime a b) (hac : is_coprime a c) : is_coprime a (b * c) := sorry
```
then `prod_right` is a simple induction.

Reply:
done